feat: filter inputs - string autocomplete to mantine

IrakliJani opened the pull request:
Closes: https://github.com/lightdash/lightdash/issues/6730

### Description:


![CleanShot 2023-10-31 at 01 26 23@2x](https://github.com/lightdash/lightdash/assets/962095/704a7bd2-9599-4178-9b12-ae0af8c47c97)

![CleanShot 2023-10-31 at 01 26 38@2x](https://github.com/lightdash/lightdash/assets/962095/28e9a087-faca-40a2-8989-8ec7ac930869)

![CleanShot 2023-10-31 at 01 27 18@2x](https://github.com/lightdash/lightdash/assets/962095/30671d15-fd32-4cc0-bf6c-b255853cc398)

![CleanShot 2023-10-31 at 01 27 31@2x](https://github.com/lightdash/lightdash/assets/962095/652152df-bde3-48df-8e5d-4b5bdfcaa36a)

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

IrakliJani commented:
@magnew I think remaining tests that fail are related to the chart config sidebar change. not sure why it's enabled here 🤔

magnew commented:
> @magnew I think remaining tests that fail are related to the chart config sidebar change. not sure why it's enabled here 🤔

My bad. I turned it on to test something. Will disable and re-run.

magnew commented on packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx:
```diff
@@ -0,0 +1,151 @@
+        { refetchOnMount: 'always' },
+    );
+
+    const results = useMemo(() => [...resultsSet], [resultsSet]);
```
No big deal, but does this useMemo do anything? It seems like it just keeps an updated copy of the results, but you could just use them directly, right? Maybe I am missing something and this is critical :D

IrakliJani replied:
oh it returns a set not an array. I think it was like that before let me check

magnew replied:
No problem, it seems ok, just a little strange

magnew commented on packages/frontend/src/components/common/Filters/FilterInputs/AutoComplete/FilterMultiAutoComplete.tsx:
```diff
@@ -0,0 +1,151 @@
+    const data = useMemo(() => {
+        // Mantine does not show value tag if value is not found in data
+        // so we need to add it manually here
+        // also we are merging status indicator as a first item
```
Isn't this what onCreate is for? This probably works too, but i think we usually add them to the data in onCreate. Which is maybe a little clearer and doesnt need an extra memo.

IrakliJani replied:
so, `data` prop is basically `options` - what options show up in the dropdown. and `value` is for an array of selected options from data. if we have 

data of = [1,2,3]
and values of = [2,3,4]
then it only displays 2,3 as tags and omits 4

hope this explanation makes sense

magnew replied:
Ok, yea, I see. Thank you.

magnew reviewed (approved):
Looks good! Left a couple non-blocking questions.

IrakliJani commented:
@magnew I was doing one more round of testing and it turns out I forgot onPaste function which I just added. I also renamed component and did some cleanups
